src/feature/followUnfollow/__tests__/followUnfollowComponent.test.js

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import { shallow } from 'enzyme';
+import React from 'react';
+import mockData from '../../../__mocks__/mockData';
+
+import { FollowUnfollowComponent } from '../FollowUnfollowComponent';
+
+const props = {
+  authorId: 2,
+  username: 'username',
+  loggedInUser: {
+    isAuthenticated: true,
+  }
+};
+
+const renderFollowUnfollow = (args) => {
+  const defaultProps = {
+    isAuthenticated: true,
+    loggedInUser: true,
+    followAuthor: jest.fn(),
+    clearFollowing: jest.fn(),
+    unfollowAuthor: jest.fn(),
+    getFollowing: jest.fn(),
+    following: {
+      map: jest.fn(),
+    },
+    history: {
+      push: jest.fn(),
+    },
+  };
+  const props = { ...defaultProps, ...args };
+  return shallow(<FollowUnfollowComponent {...props} />);
+};
+
+const wrapper = renderFollowUnfollow(props);
+const wrapper2 = renderFollowUnfollow({
+  loggedInUser: {
+    isAuthenticated: false,
+  }
+});
+
+describe('Test FollowUnfollow Component...', () => {
+  it('Should render FollowUnfollow component', () => {
+    wrapper.setProps({ following: mockData.following });
+    expect(wrapper).toHaveLength(1);
+    wrapper.instance().UNSAFE_componentWillUnmount();
+  });
+  it('should update props', () => {
+    wrapper.setProps({ isAuthenticated: false });
+    expect(wrapper).toHaveLength(1);
+  });
+
+  it('should update props', () => {
+    wrapper.setProps({ isAuthenticated: true });
+    expect(wrapper).toHaveLength(1);
+  });
+
+  test('Click Follow button...', () => {
+    const FollowButton = wrapper.find('.button');
+    FollowButton.simulate('click');
+  });
+  it('Click Follow button...', () => {
+    const FollowButton = wrapper2.find('.button');
+    FollowButton.simulate('click');
+  });
+});
```

Comment:
@raymond42 , what's the difference between the above 2 tests? The click event got simulated and nothing else! There're no assertions. You should expect some click handlers to have been got called.

Reply:
One is when the user is authenticated and another one is when the user is not authenticated.
It was not descriptive enough, I corrected it. Thanks